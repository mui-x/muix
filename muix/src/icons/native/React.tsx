import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)