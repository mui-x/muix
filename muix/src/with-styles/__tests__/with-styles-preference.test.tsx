/** @jsx platform.createElement */
import React from 'react'

import { platform } from 'reactxx-sheeter'

import { ts, traceComponentEx, ReactAny } from "reactxx-tests";

describe("WITH STYLE PREFERENCE", () => {
  it("WEB", () => traceTest(true));
  it("NATIVE", () => traceTest(false));
});

const traceTest = (isWeb: boolean) => {

  traceComponentEx(isWeb, 2,
    // sheet
    {
      root: ts.view = {
        width: 11,
        height: 11,
        maxHeight: 11,
        maxWidth: 11,
        paddingTop: 11,
        paddingLeft: 11,
        paddingRight: 11,
      },
      label: {}, webOnly: {}, nativeOnly: {},
    },
    // component code
    ({ classes, classNameX, toClassNames, styleX }) => {
      return <React.Fragment>

        <ReactAny classNameX={toClassNames([classes.root, classNameX])} styleX={styleX}>
          toClassNames([classes.root, classNameX])
        </ReactAny>

        {/*********** Reverse order in toClassNames's ruleset array *************/}
        <ReactAny classNameX={toClassNames([classNameX, classes.root])} styleX={styleX}>
          toClassNames([classNameX, classes.root])
        </ReactAny>
        
      </React.Fragment>
    },
    // component usage (Comp = withStyle(comp, <withStyle options>)
    Comp => <Comp classNameX={{ maxWidth: 33 }} classes={{ root: { paddingRight: 33 } }} styleX={{ paddingTop: 33 }} />,
    // withStyle options
    {
      defaultProps: {
        classes: { root: { width: 22 } },
        classNameX: { height: 22 },
        styleX: { maxHeight: 22, paddingTop: 22 }
      }
    })
}
