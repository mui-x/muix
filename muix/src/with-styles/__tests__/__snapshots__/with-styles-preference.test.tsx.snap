// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLE PREFERENCE NATIVE 1`] = `
<ReactAny
  style={
    Object {
      "height": 22,
      "maxHeight": 22,
      "maxWidth": 33,
      "paddingLeft": 11,
      "paddingRight": 33,
      "paddingTop": 33,
      "width": 22,
    }
  }
  trace={
    Object {
      "height": "22 /*option.classNameX*/",
      "maxHeight": "22 /*styleX*/",
      "maxWidth": "33 /*classNameX*/",
      "paddingLeft": "11 /*sheet/root*/",
      "paddingRight": "33 /*classes/root*/",
      "paddingTop": "33 /*styleX*/",
      "width": "22 /*option.classes/root*/",
    }
  }
>
  toClassNames([classes.root, classNameX])
</ReactAny>
`;

exports[`WITH STYLE PREFERENCE WEB 1`] = `
<ReactAny
  className="k i j h f e c"
  style={
    Object {
      "maxHeight": 22,
      "paddingTop": 33,
    }
  }
  trace={
    Array [
      ".k { max-width:33px /*classNameX*/ }",
      ".i { height:22px /*option.classNameX*/ }",
      ".j { padding-right:33px /*classes/root*/ }",
      ".h { width:22px /*option.classes/root*/ }",
      ".f { padding-left:11px /*sheet/root*/ }",
      ".e { padding-top:11px /*sheet/root*/ }",
      ".c { max-height:11px /*sheet/root*/ }",
    ]
  }
>
  toClassNames([classes.root, classNameX])
</ReactAny>
`;
